engine/check.py

```diff
@@ -9,6 +9,54 @@
 )
 from util.log_handler import logger
 
+def check_intersection(df_ref,df_cur):
```

Comment:
I think it would be nice to have a tiny unittest for checking if the warnings and errors are thrown as expected.